src/map.rs

```diff
@@ -216,19 +217,30 @@ impl Map {
 
         Self::draw_ui(my_id, game_state, &mut powerup_sbs, ctx, assets);
 
-        graphics::draw(ctx, &background_sb, (na::Point2::new(0., 0.),)).unwrap();
-        graphics::draw(ctx, &smoke_sb, (na::Point2::new(0., 0.),)).unwrap();
+        // the offset which causes the camera to shake upon getting hit
+        let mut hit_offset = na::Vector2::new(0., 0.);
+        if hit_effect_timer > 0. {
+            hit_offset = na::Vector2::new(
+                (hit_effect_timer*constants::HIT_SHAKE_SPEED*0.9).sin()
+                    *constants::HIT_SHAKE_MAGNITUDE*hit_effect_timer,
+                (hit_effect_timer*constants::HIT_SHAKE_SPEED*1.1).sin()
+                    *constants::HIT_SHAKE_MAGNITUDE*hit_effect_timer,
+                );
+        }
+
+        graphics::draw(ctx, &background_sb, (na::Point2::new(0., 0.) + hit_offset,)).unwrap();
+        graphics::draw(ctx, &smoke_sb, (na::Point2::new(0., 0.) + hit_offset,)).unwrap();
         for sb in plane_sbs.values() {
-            graphics::draw(ctx, sb, (na::Point2::new(0., 0.),)).unwrap();
+            graphics::draw(ctx, sb, (na::Point2::new(0., 0.) + hit_offset,)).unwrap();
```

Comment:
Why not just use effect directly since it is always added to `Point2::new(0., 0.)`.

If we expext the need for an offset in the future, we should probably change it to 

```rust
let camera_offset = hit_offset;
//...
graphics::draw(..., hit_offset);
```

Reply:
You mean `graphics::draw(..., camera_offset);`? Fixed.